src/components/ActionButtons/WatchLaterButton.jsx

```diff
@@ -0,0 +1,33 @@
+import { useUserData } from "../../contexts/providers/UserDataProvider";
+import { addToWatchLater, deleteFromWatchLater } from "../../services";
+import styles from "./ActionButton.module.css";
+import { MdOutlineWatchLater } from "react-icons/md";
+
+const WatchLaterButton = ({ video, text = false }) => {
+  const {
+    state: { watchlater },
+    dispatch: userDataDispatch,
+  } = useUserData();
+
+  const token = localStorage.getItem("token");
```

Comment:
you can do this in the AuthProvider / Auth context file.

get the token and then store it in context , and you can use useAuth to get the token anywhere in the app

Reply:
Yes, that makes more sense. I've made the changes. Thank you